### PR TITLE
kb - tips - Long running Callbacks

### PR DESCRIPTION
**Pull request by erikfTW**
This PR includes Long running callbacks tip.

Here we have the original tips: 

- [Long running Callbacks](https://www.taipy.io/tips/long-running-callbacks/)

### CODE REVIEW

File: docs/knowledge_base/tips/long_callbacks/index.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# Long Running Callbacks
+
+Tasks in web applications can take a lot of time, which can lead to problems with communication between
```
Comment (jrobinAV) [inline docs/knowledge_base/tips/long_callbacks/index.md]: I am concerned about the term "Task" here, which does not refer to the Taipy Task object. Can it be confusing for readers? @erikfTW @FabienLelaquais @FlorianJacta @RymMichaut

This comment applies to all the task usages on this page.

Alternative wording could be function, process, callback, etc.
What do you think?

File: docs/knowledge_base/tips/long_callbacks/index.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
+# Long Running Callbacks
```
Comment (jrobinAV) [inline docs/knowledge_base/tips/long_callbacks/index.md]: No need for a H1 title here. The nav bar title will be added automatically, so you will have 2 titles.
```suggestion
```

File: docs/knowledge_base/tips/long_callbacks/index.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+![Long Running Callbacks](long_running_callbacks.png){width=100%}
+
+To deal with this issue, Taipy has a helpful feature called "long-running callbacks" to address this problem. 
+These callbacks allow the server to do demanding tasks in the background while still making sure 
+the user interface stays quick to respond. 
+This article talks about the idea of long-running callbacks in Taipy, gives examples of how to use them, 
+and shows how they make the user experience better.
+
+## Example 1: Executing a Heavy Task in the Background
```
Comment (jrobinAV) [inline docs/knowledge_base/tips/long_callbacks/index.md]: If we remove the H1 title, we can change the title level accordingly.
```suggestion
# Example 1: Executing a Heavy Task in the Background
```

File: docs/knowledge_base/tips/long_callbacks/index.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+Imagine a situation where a callback starts a task that requires a lot of resources and time to finish. 
+To make this work, we can use a straightforward approach:
+
+![Executing a Heavy Task in the Background](long_running_callbacks_2.png){width=100%}
+
+In the previous example, the Taipy function called `invoke_long_callback()` manages the resource-intensive task. 
+It sets up a separate background thread to run the `heavy_function()`, allowing the rest of the application to continue running. 
+The `on_action` function gets activated by a user action, like clicking a button, for instance.
+
+## Example 2: Monitoring Task Status
```
Comment (jrobinAV) [inline docs/knowledge_base/tips/long_callbacks/index.md]:
```suggestion
# Example 2: Monitoring Task Status
```

File: docs/knowledge_base/tips/long_callbacks/index.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+Taipy offers a way to receive notifications when the task is complete, as shown below:
+
+![Monitoring Task Status](long_running_callbacks_3.png){width=100%}
+
+In this example, we introduce the `heavy_function_status()` function, which is invoked by the `invoke_long_callback()` method. 
+When the task is finished, this function is called. 
+This gives you the opportunity to provide the necessary notifications or make updates to the user interface depending 
+on whether the task was successful or not.
+
+## Handling heavy_function results: Updating the Application State
```
Comment (jrobinAV) [inline docs/knowledge_base/tips/long_callbacks/index.md]:
```suggestion
# Handling heavy_function results: Updating the Application State
```

File: docs/knowledge_base/tips/long_callbacks/index.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+Make sure that the `heavy_function()` returns a value. For example:
+
+![Updating the Application State](long_running_callbacks_5.png){width=100%}
+
+When you update the State with the result of the `heavy_function()`, you make sure that the user interface shows 
+the result of the resource-intensive task. 
+This creates a smooth and seamless user experience.
+
+## Example 3: Tracking Task Progress
```
Comment (jrobinAV) [inline docs/knowledge_base/tips/long_callbacks/index.md]:
```suggestion
# Example 3: Tracking Task Progress
```

File: docs/knowledge_base/tips/long_callbacks/index.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+Occasionally, it's useful to give regular updates on the progress of a long-running task. Taipy's `invoke_long_callback()` 
+provides a convenient method to accomplish this:
+
+![Tracking Task Progress](long_running_callbacks_6.png){width=100%}
+
+In the code above, when you include a `period` parameter, the `heavy_function_status()` function will be regularly activated 
+at the set interval, such as every 5 seconds. 
+This allows your user interface to show live updates, keeping the end user informed about the ongoing task.
+
+## Conclusion
```
Comment (jrobinAV) [inline docs/knowledge_base/tips/long_callbacks/index.md]:
```suggestion
# Conclusion
```

File: docs/knowledge_base/tips/long_callbacks/index.md (hunk 1 above)
Comment (FabienLelaquais) [reply docs/knowledge_base/tips/long_callbacks/index.md]: I completely agree.
We must find synonyms all over the place.
Here that could be replacing "Tasks in web applications" by "Executing a user-define piece of code"

File: docs/knowledge_base/tips/long_callbacks/index.md (hunk 3 above)
Comment (FabienLelaquais) [reply docs/knowledge_base/tips/long_callbacks/index.md]: May be not though.
MkDocs adds a H1-like thing, and using a H2 section makes the section title smaller, which is probably what we want.

File: docs/knowledge_base/tips/long_callbacks/index.md (hunk 4 above)
Comment (FabienLelaquais) [reply docs/knowledge_base/tips/long_callbacks/index.md]: "\`heavy_function()\`" => "\*heavy_function()\*"
"The \`on_action\` function" -> "The \`on_action\` callback"

File: docs/knowledge_base/tips/long_callbacks/index.md [hunk 8]
```diff
@@ -0,0 +1,73 @@
+# Long Running Callbacks
+
+Tasks in web applications can take a lot of time, which can lead to problems with communication between 
+the server (Python application) and the client (web browser).
+
+![Long Running Callbacks](long_running_callbacks.png){width=100%}
+
+To deal with this issue, Taipy has a helpful feature called "long-running callbacks" to address this problem.
```
Comment (FabienLelaquais) [inline docs/knowledge_base/tips/long_callbacks/index.md]: "To deal with this issue, Taipy has a helpful feature called "long-running callbacks" to address this problem."
->
"To deal with this issue, Taipy has a helpful feature called "long-running callbacks"."

File: docs/knowledge_base/tips/long_callbacks/index.md [hunk 9]
```diff
@@ -0,0 +1,73 @@
+# Long Running Callbacks
+
+Tasks in web applications can take a lot of time, which can lead to problems with communication between 
+the server (Python application) and the client (web browser).
+
+![Long Running Callbacks](long_running_callbacks.png){width=100%}
+
+To deal with this issue, Taipy has a helpful feature called "long-running callbacks" to address this problem. 
+These callbacks allow the server to do demanding tasks in the background while still making sure
```
Comment (FabienLelaquais) [inline docs/knowledge_base/tips/long_callbacks/index.md]: tasks -> processing

File: docs/knowledge_base/tips/long_callbacks/index.md [hunk 10]
```diff
@@ -0,0 +1,73 @@
+# Long Running Callbacks
+
+Tasks in web applications can take a lot of time, which can lead to problems with communication between 
+the server (Python application) and the client (web browser).
+
+![Long Running Callbacks](long_running_callbacks.png){width=100%}
+
+To deal with this issue, Taipy has a helpful feature called "long-running callbacks" to address this problem. 
+These callbacks allow the server to do demanding tasks in the background while still making sure 
+the user interface stays quick to respond.
```
Comment (FabienLelaquais) [inline docs/knowledge_base/tips/long_callbacks/index.md]: "quick to respond" -> "responsive"

File: docs/knowledge_base/tips/long_callbacks/index.md [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+
+Tasks in web applications can take a lot of time, which can lead to problems with communication between 
+the server (Python application) and the client (web browser).
+
+![Long Running Callbacks](long_running_callbacks.png){width=100%}
+
+To deal with this issue, Taipy has a helpful feature called "long-running callbacks" to address this problem. 
+These callbacks allow the server to do demanding tasks in the background while still making sure 
+the user interface stays quick to respond. 
+This article talks about the idea of long-running callbacks in Taipy, gives examples of how to use them,
```
Comment (FabienLelaquais) [inline docs/knowledge_base/tips/long_callbacks/index.md]: idea -> concept

File: docs/knowledge_base/tips/long_callbacks/index.md [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+Tasks in web applications can take a lot of time, which can lead to problems with communication between 
+the server (Python application) and the client (web browser).
+
+![Long Running Callbacks](long_running_callbacks.png){width=100%}
+
+To deal with this issue, Taipy has a helpful feature called "long-running callbacks" to address this problem. 
+These callbacks allow the server to do demanding tasks in the background while still making sure 
+the user interface stays quick to respond. 
+This article talks about the idea of long-running callbacks in Taipy, gives examples of how to use them, 
+and shows how they make the user experience better.
```
Comment (FabienLelaquais) [inline docs/knowledge_base/tips/long_callbacks/index.md]: "make the user experience better." -> "can improve the user experience."

File: docs/knowledge_base/tips/long_callbacks/index.md [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+
+## Example 1: Executing a Heavy Task in the Background
+
+Imagine a situation where a callback starts a task that requires a lot of resources and time to finish.
```
Comment (FabienLelaquais) [inline docs/knowledge_base/tips/long_callbacks/index.md]: task -> duty

File: docs/knowledge_base/tips/long_callbacks/index.md [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+the user interface stays quick to respond. 
+This article talks about the idea of long-running callbacks in Taipy, gives examples of how to use them, 
+and shows how they make the user experience better.
+
+## Example 1: Executing a Heavy Task in the Background
+
+Imagine a situation where a callback starts a task that requires a lot of resources and time to finish. 
+To make this work, we can use a straightforward approach:
+
+![Executing a Heavy Task in the Background](long_running_callbacks_2.png){width=100%}
```
Comment (FabienLelaquais) [inline docs/knowledge_base/tips/long_callbacks/index.md]: "a Heavy Task" -> "a time-consuming work"

File: docs/knowledge_base/tips/long_callbacks/index.md [hunk 15]
```diff
@@ -0,0 +1,73 @@
...
+
+In the previous example, the Taipy function called `invoke_long_callback()` manages the resource-intensive task.
```
Comment (FabienLelaquais) [inline docs/knowledge_base/tips/long_callbacks/index.md]: \`invoke_long_callback()\` => \`invoke_long_callback()^\`

File: docs/knowledge_base/tips/long_callbacks/index.md [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+
+![Executing a Heavy Task in the Background](long_running_callbacks_2.png){width=100%}
+
+In the previous example, the Taipy function called `invoke_long_callback()` manages the resource-intensive task. 
+It sets up a separate background thread to run the `heavy_function()`, allowing the rest of the application to continue running. 
+The `on_action` function gets activated by a user action, like clicking a button, for instance.
+
+## Example 2: Monitoring Task Status
+
+Moreover, you can send notifications to the user's browser or update visual elements depending on the task's result.
```
Comment (FabienLelaquais) [inline docs/knowledge_base/tips/long_callbacks/index.md]: "task's result" -> "status of the ongoing process"

File: docs/knowledge_base/tips/long_callbacks/index.md [hunk 17]
```diff
@@ -0,0 +1,73 @@
...
+The `on_action` function gets activated by a user action, like clicking a button, for instance.
+
+## Example 2: Monitoring Task Status
+
+Moreover, you can send notifications to the user's browser or update visual elements depending on the task's result. 
+Taipy offers a way to receive notifications when the task is complete, as shown below:
+
+![Monitoring Task Status](long_running_callbacks_3.png){width=100%}
+
+In this example, we introduce the `heavy_function_status()` function, which is invoked by the `invoke_long_callback()` method.
```
Comment (FabienLelaquais) [inline docs/knowledge_base/tips/long_callbacks/index.md]: \`heavy_function_status()\` => \*heavy_function_status()\*
\`invoke_long_callback()\` => \`invoke_long_callback()^\`

File: docs/knowledge_base/tips/long_callbacks/index.md [hunk 18]
```diff
@@ -0,0 +1,73 @@
...
+
+## Example 2: Monitoring Task Status
+
+Moreover, you can send notifications to the user's browser or update visual elements depending on the task's result. 
+Taipy offers a way to receive notifications when the task is complete, as shown below:
+
+![Monitoring Task Status](long_running_callbacks_3.png){width=100%}
+
+In this example, we introduce the `heavy_function_status()` function, which is invoked by the `invoke_long_callback()` method. 
+When the task is finished, this function is called.
```
Comment (FabienLelaquais) [inline docs/knowledge_base/tips/long_callbacks/index.md]: task -> callback
Add \<br/\> before the line break.

File: docs/knowledge_base/tips/long_callbacks/index.md [hunk 19]
```diff
@@ -0,0 +1,73 @@
...
+
+Moreover, you can send notifications to the user's browser or update visual elements depending on the task's result. 
+Taipy offers a way to receive notifications when the task is complete, as shown below:
+
+![Monitoring Task Status](long_running_callbacks_3.png){width=100%}
+
+In this example, we introduce the `heavy_function_status()` function, which is invoked by the `invoke_long_callback()` method. 
+When the task is finished, this function is called. 
+This gives you the opportunity to provide the necessary notifications or make updates to the user interface depending 
+on whether the task was successful or not.
```
Comment (FabienLelaquais) [inline docs/knowledge_base/tips/long_callbacks/index.md]: task -> processing

**Comment by jrobinAV** [comment]
Sorry, I deleted the branch by mistake. I reopened it and re-opened the PR as well.

File: docs/knowledge_base/tips/long_callbacks/index.md [hunk 20]
```diff
@@ -0,0 +1,73 @@
...
+
+## Handling heavy_function results: Updating the Application State
+
+To update the State according to the return of the `heavy_function`, you can modify the `heavy_function_status()` function as follows:
```
Comment (FabienLelaquais) [inline docs/knowledge_base/tips/long_callbacks/index.md]: "the return of the \`heavy_function\`, you can modify the \`heavy_function_status()\` function as follows:"
->
"the returned value from \*heavy_function()\*, you can modify \*heavy_function_status()\* as follows:"

File: docs/knowledge_base/tips/long_callbacks/index.md [hunk 21]
```diff
@@ -0,0 +1,73 @@
...
+
+![Updating the Application State](long_running_callbacks_4.png){width=100%}
+
+We added a parameter called result, which represents the return value of the `heavy_function()`. When the `heavy_function()`
```
Comment (FabienLelaquais) [inline docs/knowledge_base/tips/long_callbacks/index.md]: result -> \*result\*
"the \`heavy_function()\`" -> "\*heavy_function()\*" (twice!)

File: docs/knowledge_base/tips/long_callbacks/index.md [hunk 22]
```diff
@@ -0,0 +1,73 @@
...
+on whether the task was successful or not.
+
+## Handling heavy_function results: Updating the Application State
+
+To update the State according to the return of the `heavy_function`, you can modify the `heavy_function_status()` function as follows:
+
+![Updating the Application State](long_running_callbacks_4.png){width=100%}
+
+We added a parameter called result, which represents the return value of the `heavy_function()`. When the `heavy_function()` 
+completes successfully (status is True),
```
Comment (FabienLelaquais) [inline docs/knowledge_base/tips/long_callbacks/index.md]: status -> \*status\*

File: docs/knowledge_base/tips/long_callbacks/index.md [hunk 23]
```diff
@@ -0,0 +1,73 @@
...
+
+## Handling heavy_function results: Updating the Application State
+
+To update the State according to the return of the `heavy_function`, you can modify the `heavy_function_status()` function as follows:
+
+![Updating the Application State](long_running_callbacks_4.png){width=100%}
+
+We added a parameter called result, which represents the return value of the `heavy_function()`. When the `heavy_function()` 
+completes successfully (status is True), 
+we update the State with the result by assigning it to a state variable (e.g., state.result = result).
```
Comment (FabienLelaquais) [inline docs/knowledge_base/tips/long_callbacks/index.md]: "state.result = result" -> "\`state.result = result\`"

File: docs/knowledge_base/tips/long_callbacks/index.md [hunk 24]
```diff
@@ -0,0 +1,73 @@
...
+To update the State according to the return of the `heavy_function`, you can modify the `heavy_function_status()` function as follows:
+
+![Updating the Application State](long_running_callbacks_4.png){width=100%}
+
+We added a parameter called result, which represents the return value of the `heavy_function()`. When the `heavy_function()` 
+completes successfully (status is True), 
+we update the State with the result by assigning it to a state variable (e.g., state.result = result). 
+This allows you to access the result in other parts of your application or display it to the user as needed.
+
+Make sure that the `heavy_function()` returns a value. For example:
```
Comment (FabienLelaquais) [inline docs/knowledge_base/tips/long_callbacks/index.md:80]: "the \`heavy_function()\`" -> "*heavy_function()*"

File: docs/knowledge_base/tips/long_callbacks/index.md [hunk 25]
```diff
@@ -0,0 +1,73 @@
...
+
+![Updating the Application State](long_running_callbacks_5.png){width=100%}
+
+When you update the State with the result of the `heavy_function()`, you make sure that the user interface shows
```
Comment (FabienLelaquais) [inline docs/knowledge_base/tips/long_callbacks/index.md]: "the \`heavy_function()\`" -> "*heavy_function()*"

File: docs/knowledge_base/tips/long_callbacks/index.md [hunk 26]
```diff
@@ -0,0 +1,73 @@
...
+
+![Updating the Application State](long_running_callbacks_5.png){width=100%}
+
+When you update the State with the result of the `heavy_function()`, you make sure that the user interface shows 
+the result of the resource-intensive task. 
+This creates a smooth and seamless user experience.
+
+## Example 3: Tracking Task Progress
+
+Occasionally, it's useful to give regular updates on the progress of a long-running task. Taipy's `invoke_long_callback()`
```
Comment (FabienLelaquais) [inline docs/knowledge_base/tips/long_callbacks/index.md]: \`invoke_long_callback()\` => \`invoke_long_callback()^\`

File: docs/knowledge_base/tips/long_callbacks/index.md [hunk 27]
```diff
@@ -0,0 +1,73 @@
...
+This creates a smooth and seamless user experience.
+
+## Example 3: Tracking Task Progress
+
+Occasionally, it's useful to give regular updates on the progress of a long-running task. Taipy's `invoke_long_callback()` 
+provides a convenient method to accomplish this:
+
+![Tracking Task Progress](long_running_callbacks_6.png){width=100%}
+
+In the code above, when you include a `period` parameter, the `heavy_function_status()` function will be regularly activated
```
Comment (FabienLelaquais) [inline docs/knowledge_base/tips/long_callbacks/index.md]: \`period\` -> \*period\*
\`heavy_function_status()\` -> \*heavy_function_status()\*

File: docs/knowledge_base/tips/long_callbacks/index.md [hunk 28]
```diff
@@ -0,0 +1,73 @@
...
+## Example 3: Tracking Task Progress
+
+Occasionally, it's useful to give regular updates on the progress of a long-running task. Taipy's `invoke_long_callback()` 
+provides a convenient method to accomplish this:
+
+![Tracking Task Progress](long_running_callbacks_6.png){width=100%}
+
+In the code above, when you include a `period` parameter, the `heavy_function_status()` function will be regularly activated 
+at the set interval, such as every 5 seconds. 
+This allows your user interface to show live updates, keeping the end user informed about the ongoing task.
```
Comment (FabienLelaquais) [inline docs/knowledge_base/tips/long_callbacks/index.md]: task -> work

**Review by FabienLelaquais — DISMISSED** [review]
A few minor things, including conventions on typography.

**Comment by FabienLelaquais** [comment]
I sincerely hope that the code fragments on the web site (where, well, it wasn't perfect) are kept!
It looks like images replaced code blocks, which makes it impossible to copy/paste it.

File: docs/knowledge_base/tips/long_callbacks/index.md (hunk 1 above)
Comment (FlorianJacta) [reply docs/knowledge_base/tips/long_callbacks/index.md]: Agree! However, this is the terminology of the [doc](https://docs.taipy.io/en/develop/manuals/gui/callbacks/#long-running-callbacks). It should be changed there as well. The goal was to be consistent with the user man.

Comment (jrobinAV) [reply docs/knowledge_base/tips/long_callbacks/index.md]: A new ticket was created on taipy-doc for user manual:
https://github.com/Avaiga/taipy-doc/issues/603
We can proceed and apply the change on this page. We will take care of the user manual later.

File: docs/knowledge_base/tips/long_callbacks/index.md (hunk 3 above)
Comment (jrobinAV) [reply docs/knowledge_base/tips/long_callbacks/index.md]: Oh... I created dozens of comments in other PRs about that.
🤣🤣🤣🤣
 
We need to make a global decision, I guess.

File: docs/knowledge_base/tips/long_callbacks/index.md (hunk 4 above)
Comment (jrobinAV) [reply docs/knowledge_base/tips/long_callbacks/index.md]: Wrong line comment, no?

**Comment by erikfTW** [comment]
> I sincerely hope that the code fragments on the web site (where, well, it wasn't perfect) are kept! It looks like images replaced code blocks, which makes it impossible to copy/paste it.

I have fixed that and added to code blocks.

**Review by jrobinAV — APPROVED** [review]

**Comment by jrobinAV** [comment]
Please review it again. I applied doc conventions + I tried changing the wording on Task.
Not sure abt the result.

**Comment by RymMichaut** [comment]
I feel like this article lacks of explanatory images

File: docs/knowledge_base/tips/long_callbacks/index.md [hunk 29]
```diff
@@ -0,0 +1,123 @@
...
+Tasks (server-side functions) in web applications can take a lot of time, which can lead to 
+problems with communication between the server (Python application) and the client (web browser).
+
+Taipy offers a valuable feature known as "long-running callbacks" to tackle this problem. These 
+callbacks enable the server to handle resource-intensive processing in the background, ensuring 
+a responsive user interface. 
+
+![Long Running Callbacks](long_running_callbacks.png){width=100%}
+
+This article discusses the concept of long-running callbacks in Taipy, provides usage examples,
```
Comment (FlorianJacta) [inline docs/knowledge_base/tips/long_callbacks/index.md:11]: tip

**Pull request merged**